common/src/main/java/com/wynntils/screens/maps/WaypointCategoryScreen.java

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.maps;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.screens.base.WynntilsGridLayoutScreen;
+import com.wynntils.screens.base.widgets.TextInputBoxWidget;
+import com.wynntils.screens.maps.widgets.CategoryWidget;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+
+public final class WaypointCategoryScreen extends WynntilsGridLayoutScreen {
+    private static final int CATEGORIES_PER_PAGE = 16;
+    private static final int SCROLLBAR_HEIGHT = 40;
+    private static final float SCROLL_FACTOR = 10f;
+    private static final String DEFAULT_CATEGORY = "wynntils:personal:waypoint";
+    private static final Pattern CATEGORY_PATTERN = Pattern.compile("^[a-zA-Z0-9:-]+$");
```

Comment:
I think I told you A-Z is fine, but maybe we shouldn't do that. a-z0-9:- should be enough.